alpenhorn/io/ioutil.py

```diff
@@ -314,6 +316,54 @@ def hardlink(from_path, to_dir, filename) -> dict | None:
     return {"ret": 0, "md5sum": True}
 
 
+def post_add(node: StorageNode, file_: ArchiveFile) -> None:
+    """Run any actions after adding `file` to `node`.
+
+    Possible actions are autosync or autoclean.
+
+    Parameters
+    ----------
+    node : StorageNode
+        The node the file copy was added to.
+    file_ : ArchiveFile
+        The file added.
+    """
+
+    # Autosync: find all StorageTransfers where we're the source node
+    for edge in StorageTransfer.select().where(
+        StorageTransfer.node_from == node,
+        StorageTransfer.group_to != node.group,
+        StorageTransfer.autosync == True,
+    ):
+        if edge.group_to.filecopy_state(file_) == "N":
+            log.debug(
+                f"Autosyncing {file_.path} from node {node.name} to group {edge.group_to.name}"
+            )
+
+            ArchiveFileCopyRequest.create(
+                node_from=node, group_to=edge.group_to, file=file_
+            )
+
+    # Autoclean: find all the StorageTransfers where we're in the dest group
+    for edge in StorageTransfer.select().where(
+        StorageTransfer.group_to == node.group,
+        StorageTransfer.node_from != node,
+        StorageTransfer.autoclean == True,
+    ):
+        try:
+            copy = ArchiveFileCopy.get(
```

Comment:
Could you just update all relevant file copies in here to save a query? Rather than the .get and .save combo.